### PR TITLE
Rename 'select by form' to 'select by value'

### LINKED ISSUE #1124: [needs-docs] Rename 'select by form' to 'select by value'
Original commit: https://github.com/qgis/QGIS/commit/9a06144819eee30ebb7d3c45bfe8626fd16886d5 by nyalldawson

As that wording is more explanatory

### PR DESCRIPTION
**Pull request by yjacolin**
Fix #1124. It is mainly an update of the list of select features available in the menu.

Funded by camptocamp

### CODE REVIEW

File: source/docs/user_manual/introduction/general_tools.rst [hunk 1]
```diff
@@ -484,12 +484,16 @@ tool:
 * |selectFreehand| :sup:`Select Features by Freehand`
 * |selectRadius| :sup:`Select Features by Radius`
 
-To deselect all selected features, click on |deselectAll| :sup:`Deselect
-Features from All Layers`.
-
-|expressionSelect| :sup:`Select features using an expression` allows user
-to select features using expression dialog. See :ref:`vector_expressions`
-chapter for some example.
+The :menuselection:`View --> Selection` menu adds more possibilities:
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: View --> Select

Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: Why refering to this menu given that those tools are also available in the toolbar (in the drop-down list of `select by expression`)?

**Comment by DelazJ** [comment]
Wouldn't it be interesting to present the `select features by value` dialog? It has a lot of options that may worth it. I don't know if this chapter is the more appropriate though...

File: source/docs/user_manual/introduction/general_tools.rst (hunk 1 above)
Comment (yjacolin) [inline source/docs/user_manual/introduction/general_tools.rst]: arg, did'nt see them in the toolbar. I rephrase this section.

File: source/docs/user_manual/introduction/general_tools.rst [hunk 2]
```diff
@@ -484,17 +490,52 @@ tool:
...
-|expressionSelect| :sup:`Select features using an expression` allows user
-to select features using expression dialog. See :ref:`vector_expressions`
-chapter for some example.
+* |expressionSelect| :sup:`Select By Expression...` allows user to select
+  features using expression dialog. See :ref:`vector_expressions` chapter for
+  some example.
+* |formSelect| :sup:`Select Features By Value...` or press :kbd:`F3`
+* |deselectAll| :sup:`Deselect Features from All Layers` or press
+  :kbd:`Ctrl+Shift+A` to deselect all selected features.
+* |selectAll| :sup:`Select All Features` or press :kbd:`Ctrl+A` to select all features.
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: maybe, could we add `in the current layer` (to show that you `deselect all` from all layers but `select all` only within the selected layer

File: source/docs/user_manual/introduction/general_tools.rst [hunk 3]
```diff
@@ -484,17 +490,52 @@ tool:
...
+.. note:: The :menuselection:`View --> Select` menu list the same possibilities.
 
 Users can save selected features into a **New Memory Vector Layer** or a
 **New Vector Layer** using :menuselection:`Edit --> Copy Features` and
 :menuselection:`Edit --> Paste Features as` in the wanted format.
 
+.. index:: Select; By Value
+
+About Select Features By Value
+*******************************
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: According to [guidelines](http://docs.qgis.org/2.14/en/docs/documentation_guidelines/writing.html#headlines), it must be `------` style for consistency

File: source/docs/user_manual/introduction/general_tools.rst [hunk 4]
```diff
@@ -484,17 +490,52 @@ tool:
...
+
+This feature open a windows to allow the user to choose for each column of the
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: opens a window

File: source/docs/user_manual/introduction/general_tools.rst [hunk 5]
```diff
@@ -484,17 +490,52 @@ tool:
...
-to select features using expression dialog. See :ref:`vector_expressions`
-chapter for some example.
+* |expressionSelect| :sup:`Select By Expression...` allows user to select
+  features using expression dialog. See :ref:`vector_expressions` chapter for
+  some example.
+* |formSelect| :sup:`Select Features By Value...` or press :kbd:`F3`
+* |deselectAll| :sup:`Deselect Features from All Layers` or press
+  :kbd:`Ctrl+Shift+A` to deselect all selected features.
+* |selectAll| :sup:`Select All Features` or press :kbd:`Ctrl+A` to select all features.
+* |invertSelection| :sup:`Invert Feature Selection` to invert the selection.
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: also in the current layer

File: source/docs/user_manual/introduction/general_tools.rst (hunk 2 above)
Comment (yjacolin) [inline source/docs/user_manual/introduction/general_tools.rst]: yes, added

File: source/docs/user_manual/introduction/general_tools.rst (hunk 5 above)
Comment (yjacolin) [inline source/docs/user_manual/introduction/general_tools.rst]: added.

File: source/docs/user_manual/introduction/general_tools.rst [hunk 6]
```diff
@@ -469,6 +469,12 @@ is displayed in a pop-up dialog.
 
    Measure Angle
 
+.. index::
+   single: Select
+   single: Select; All, Select; Invert, Select; Expression, Select: Form
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: Why a `:` and not a `;`?

File: source/docs/user_manual/introduction/general_tools.rst [hunk 7]
```diff
@@ -469,6 +469,12 @@ is displayed in a pop-up dialog.
...
+   single: Select; Polygon, Select; Freehand, Select; Rectangle, Select; Radus
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: Radius

File: source/docs/user_manual/introduction/general_tools.rst (hunk 6 above)
Comment (yjacolin) [inline source/docs/user_manual/introduction/general_tools.rst]: it is a typo, thanks

**Comment by DelazJ** [comment]
added some comments otherwise LGTM

**Comment by yjacolin** [comment]
Waiting for the test. I commit my changes in regards to your comments. Thanks.

**Pull request merged**